e2e/scripts/st_tooltips.py

```diff
@@ -1,77 +0,0 @@
-# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-from datetime import datetime
-
-import streamlit as st
-
-default_tooltip = """
-This is a really long tooltip.
-
-Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut ut turpis vitae
-justo ornare venenatis a vitae leo. Donec mollis ornare ante, eu ultricies
-tellus ornare eu. Donec eros risus, ultrices ut eleifend vel, auctor eu turpis.
-In consectetur erat vel ante accumsan, a egestas urna aliquet. Nullam eget
-sapien eget diam euismod eleifend. Nulla purus enim, finibus ut velit eu,
-malesuada dictum nulla. In non arcu et risus maximus fermentum eget nec ante.
-""".strip()
-
-leading_indent_code_tooltip = """
-Code:
-
-    This
-    is
-    a
-    code
-    block!"""
-
-leading_indent_regular_text_tooltip = """
-This is a regular text block!
-Test1
-Test2
-
-"""
-
-indented_code_tooltip = """
-Code:
-
-    for i in range(10):
-        x = i * 10
-        print(x)
-    """
-
-no_indent_tooltip = "thisisatooltipwithnoindents. It has some spaces but no idents."
-
-st.text_input("some input text", "default text", help=default_tooltip)
```

Comment:
I think we should add a test case to every individual widget e2e test to make sure that the hovering works. 
I think it's totally fine to do that for the widgets with a simple text like "help text" or whatever is currently used in the script -> main goal would be to confirm that hovering works and shows a tooltip the correct text.

Reply:
Hm wouldn't adding every individual widget simply use the same Tooltip component and that would be redundant testing?

Reply:
Not really. Just having tooltip tests for a few widgets won't guard against the scenario where tooltip code just isn't set up correctly for a given widget

Reply:
Can you give an example of what you're speaking of? From my perspective, won't widgets take up the container size and then the tooltip will be attached that?

Reply:
Its a bit redundant, but I think it's still reasonable that every element has an associated test that makes sure that this functionality works end to end (setting the `help` provides a hoverable help tooltip). The element implementation still needs to include and correctly implement our shared help component.

But the good thing is that it will only require the  same 5 lines to be added to every element with help tooltip to properly test this (for all elements that use the shared tooltip impl.:

(needs this PR merged: https://github.com/streamlit/streamlit/pull/8780/files)

```
from e2e_playwright.shared.app_utils import expect_help_tooltip

....

def test_help_tooltip_works(app: Page):
    """Test that the help tooltip is displayed on hover."""
    element_with_help = app.get_by_test_id("stTextInput").nth(8)
    expect_help_tooltip(app, element_with_help, "Help text")
```